utils/python/CIME/case_st_archive.py

```diff
@@ -132,6 +132,7 @@ def _archive_history_files(case, archive, archive_entry,
 
     # determine history archive directory (create if it does not exist)
     dout_s_root = case.get_value("DOUT_S_ROOT")
+    casename = case.get_value("CASE")
```

Comment:
by explicitly including the casename we prevent suffixes matching something in that part of the string, this fixes #523